OADP-6673: Add OADP 1.6.0 RNs (#112544)
openshift/openshift-docs: modules/oadp-1-6-0-release-notes.adoc

+
link:https://redhat.atlassian.net/browse/OADP-2785[OADP-2785]

[id="resolved-issues-1-6-0_{context}"]
== Resolved issues

Velero backup with `labelSelector` no longer has degraded performance during resource discovery::
Before this update, a performance regression in the resource discovery phase caused Velero to issue an excessive number of API list calls when a backup combined `includedNamespaces: '*'`, `includeClusterResources: true`, `labelSelector`, and a large `excludedResources` list. As a consequence, backups that completed before in seconds took about 18 minutes to back up as few as 8 resources. With this release, the resource discovery logic is optimized so that unnecessary API list calls are eliminated during the item collection phase. As a result, backup performance is restored to expected levels, completing in seconds.
+
link:https://redhat.atlassian.net/browse/OADP-7977[OADP-7977]

CSI snapshot early frequent polling can be configured in the DPA CR::

> Before this update, a Velero change introduced 1-second Container Storage Interface (CSI) snapshot readiness polling for the first 10 seconds of every backup, designed specifically for Microsoft Volume Shadow Copy Service (VSS) workloads on Windows. {oadp-short} did not expose this behavior as a configurable option in the `DataProtectionApplication` (DPA) custom resource (CR). As a consequence, all CSI-enabled backups used the earlier polling behavior regardless of whether VSS compatibility was needed, and you had no way to disable or control this behavior through {oadp-short}. This release adds a new `disableCSISnapshotEarlyFrequentPolling` field under `configuration.velero` in the DPA CR. As a result, you can tailor CSI snapshot polling behavior to your workload requirements directly from the DPA configuration.

Reviewer: the field name is changed and by default its disabled. Users needs to enable it 

enableCSISnapshotEarlyFrequentPolling

Author: Thank you @PrasadJoshi12, I will update the text in my next commit.

FROM:
CSI snapshot early frequent polling can be configured in the DPA CR::
Before this update, a Velero change introduced 1-second Container Storage Interface (CSI) snapshot readiness polling for the first 10 seconds of every backup, designed specifically for Microsoft Volume Shadow Copy Service (VSS) workloads on Windows. {oadp-short} did not expose this behavior as a configurable option in the `DataProtectionApplication` (DPA) custom resource (CR). As a consequence, all CSI-enabled backups used the earlier polling behavior regardless of whether VSS compatibility was needed, and you had no way to disable or control this behavior through {oadp-short}. This release adds a new `disableCSISnapshotEarlyFrequentPolling` field under `configuration.velero` in the DPA CR. As a result, you can tailor CSI snapshot polling behavior to your workload requirements directly from the DPA configuration.

TO:
CSI snapshot early frequent polling can be configured in the DPA CR::
Before this update, a Velero change introduced 1-second Container Storage Interface (CSI) snapshot readiness polling for the first 10 seconds of every backup, designed specifically for Microsoft Volume Shadow Copy Service (VSS) workloads on Windows. {oadp-short} did not expose this behavior as a configurable option in the `DataProtectionApplication` (DPA) custom resource (CR). As a consequence, all CSI-enabled backups used the earlier polling behavior regardless of whether VSS compatibility was needed, and you had no way to disable or control this behavior through {oadp-short}. This release adds a new `enableCSISnapshotEarlyFrequentPolling` field under `configuration.velero` in the DPA CR. By default, this feature is disabled. If your workloads require this frequent polling behavior, you must enable it.

+
link:https://redhat.atlassian.net/browse/OADP-7132[OADP-7132]

`PrepareQueueLength` is present in DPA::
The `PrepareQueueLength` field is a configuration setting for the node agent that limits the number of intermediate resources created during backup and restore processes. Before this update, `PrepareQueueLength` was missing in the `LoadConcurrency` field in the `DataProtectionApplication` (DPA) configuration. As a consequence, you encountered an error while configuring `PrepareQueueLength`, causing the configuration to be removed. With this release, the `PrepareQueueLength` field is added to `LoadConcurrency` in the DPA configuration.
+
link:https://redhat.atlassian.net/browse/OADP-7070[OADP-7070]

Restore transitions to `Completed` only after `PodVolumeRestores` completes::
Before this update, restores were marked as completed prematurely before all `PodVolumeRestores`. With this release, restores complete only after `PodVolumeRestores`.
+
link:https://redhat.atlassian.net/browse/OADP-6911[OADP-6911]

> Improved volume policy performance for clusters with many pods and PVCs::

Reviewer 2: The "Improved volume policy performance" resolved issue links to OADP-6896, but that Jira is about node-agent pod restarts canceling DataUploads. The correct reference should be OADP-6879 (volume policy performance). This also means OADP-6896 itself is missing a release notes entry -- it's a Major bug about DataUploads being canceled across all nodes when a single node-agent restarts.

----
+
To work around this problem, copy `ceph-csi-kms-token` and the associated Key Management Service (KMS) config map from the application namespace to `openshift-adp` before running the backup. As a result, the ceph-csi driver can locate the encryption credentials and the backup completes successfully.
+
link:https://redhat.atlassian.net/browse/OADP-7972[OADP-7972]

The Velero server might be terminated with an `OOMKilled` event when accessing large Kopia repositories::
When the Velero server accesses large Kopia repositories, for example, during backup deletion, the Velero server might be terminated by the Out Of Memory (OOM) process. This is typically due to high memory usage loading many Kopia index blobs and related repository metadata.
+
--
To work around this problem, perform the following actions:

> . Check Velero logs around the `OOMKilled` event for repository-open or index-loading activity, and note the approximate maximum memory usage observed for the Velero pod (from oc orkubectl top pods or similar).

Reviewer 2: "oc orkubectl" should be "oc or kubectl"

. Increase the Velero container memory request or limit in `spec.velero.podConfig` in the `DataProtectionApplication` custom resource to exceed that observed peak.
. Redeploy the Velero pod until `OOMKilled` events no longer occur.
. Record the final stable memory values for your environment for future sizing guidance.
--
+
As a result, the Velero server has enough memory to load the repository metadata without being terminated.
+
link:https://redhat.atlassian.net/browse/OADP-6736[OADP-6736]

> 

Reviewer 2: A few resolved bugs with fixVersion 1.6.0 are missing from the release notes. The ones I'd call out as needing entries:
  - OADP-6955 (Critical): BackupRepository name > 63 chars causes Job build error (from ROSA HCP testing)
  - OADP-6689 (Major): Backups stuck in "Deleting" state after TTL expires
  - OADP-7381 (Critical): VGS restore fails with Ceph RBD storage class
  - OADP-3378: Misleading certificate error log when cert check is disabled

Reviewer: OADP-6689 closed with resolution cannot reproduce

Author: @shubham-pampattiwar, @PrasadJoshi12, indeed OADP-6689 is closed as Cannot reproduce.
OADP-7381 looks like it wasn't tested for 1.6.0 since VGS is Technology Preview in 1.6.0.
OADP-3378 says in the last comment from Aziza that "the insecureSkipTLSVerify path still produces the error because the CLI does not inherit this setting from the BSL".

If all above is true, they do not look like Release Note candidates for 1.6.0. Could you verify? Thank you.

I'll address the rest of feedback with a new commit.

[id="technology-preview-features-1-6-0_{context}"]
== Technology Preview features

> // Q: what is the official name for this feature?

Reviewer 2: Shouldn't this be removed ? // Q: what is the official name for this feature?

//
// * backup_and_restore/application_backup_and_restore/release-notes/oadp-1-6-release-notes.adoc
:_mod-docs-content-type: REFERENCE

[id="oadp-1-6-0-release-notes_{context}"]
= {oadp-short} 1.6.0 release notes

[role="_abstract"]
Review new features and enhancements, resolved issues, known issues, Technology Preview features, and removed features in {oadp-first} 1.6.0.

[id="new-features-1-6-0_{context}"]
== New features

Parallel backup processing::
{oadp-short} supports parallel backup processing. Before this update, {oadp-short} processed backups sequentially where each backup had to complete before the next one could begin. This created bottlenecks for environments with many backup schedules or large-scale data protection needs, increasing the overall time required to complete all backup operations. With this release, multiple backups can run in parallel if they do not share namespaces. As a result, you can now define more granular or frequent backup schedules without worrying about queuing delays.
+
link:https://redhat.atlassian.net/browse/OADP-6579[OADP-6579]

File-level restore with VMFR from KubeVirt virtual machine backups::

> {oadp-short} introduces the virtual machine file restore (VMFR) feature. Before this release, restoring even a single file from a KubeVirt VM backup required restoring the entire VM, which was time-consuming and resource-intensive. With this release, you can selectively recover individual files from KubeVirt VM backups directly within OpenShift. VMFR also includes `RestoreItemAction`, a plugin for collision prevention in multi-backup scenarios. This plugin renames PVCs during restore, ensuring unique names and avoiding conflicts, thereby improving the reliability and efficiency of VMFR backups and restores.

Reviewer 3: ```suggestion
{oadp-short} introduces the virtual machine file restore (VMFR) feature. Before this release, restoring even a single file from a KubeVirt VM backup required restoring the entire VM, which was time-consuming and resource-intensive. With this release, you can selectively recover individual files from KubeVirt VM backups directly within {product-title}. VMFR also includes `RestoreItemAction`, a plugin for collision prevention in multi-backup scenarios. This plugin renames PVCs during restore, ensuring unique names and avoiding conflicts, thereby improving the reliability and efficiency of VMFR backups and restores.
```

+
link:https://redhat.atlassian.net/browse/OADP-6583[OADP-6583]

{oadp-short} CLI plugin::
{oadp-short} includes a `kubectl` command-line interface (CLI) plugin that provides a `kubectl` native interface for managing backups and restores. Before this update, you aliased to the Velero CLI to perform backup operations. Non-cluster administrators had no way to independently manage their own backups, creating a dependency on cluster admins for routine data protection tasks. As a result, admins can run `kubectl oadp` commands to create, delete, and retrieve logs for backups and restores. Non-cluster admins can independently create, delete, and inspect non-admin backups within their permitted namespaces.
+
link:https://redhat.atlassian.net/browse/OADP-6544[OADP-6544]

{oadp-short} must-gather collects VM file restore CRDs::

> With this enhancement, the oadp-must-gather tool collects the `VirtualMachineBackupsDiscovery` and `VirtualMachineFileRestore` custom resource definitions (CDRs) introduced in {oadp-short} 1.6 for the virtual machine file-level restore feature. As a result, when you run the `oc adm must-gather` command, both new resource types are automatically collected cluster-wide, including CRDs, per-namespace CR YAMLs, and dedicated sections in the markdown summary. This gives support teams and administrators full visibility into VM file restore operations.

Reviewer 3: ```suggestion
With this enhancement, the oadp-must-gather tool collects the `VirtualMachineBackupsDiscovery` and `VirtualMachineFileRestore` custom resource definitions (CRDs) introduced in {oadp-short} 1.6 for the virtual machine file-level restore feature. As a result, when you run the `oc adm must-gather` command, both new resource types are automatically collected cluster-wide, including CRDs, per-namespace CR YAMLs, and dedicated sections in the markdown summary. This gives support teams and administrators full visibility into VM file restore operations.
```

+
link:https://redhat.atlassian.net/browse/OADP-7872[OADP-7872]

Backup and restore of custom SCC through `ClusterRole` and `ClusterRoleBinding`::

> {oadp-short} supports backup and restore of custom security context constraints (SCC) with `ServiceAccounts` through `ClusterRole` and `ClusterRoleBinding` resources, not only those listed in the SCC user fields. This enhancement simplifies the backup and restore process of custom SCCs, as it no longer relies on the user field of the SCC. Instead, it uses the location of the SCC used by a pod and the service account associated with the pod. This results in a more reliable and secure backup and restore process for OpenShift clusters, essential for IBM Cloud Pak for Data.

Reviewer 3: ```suggestion
{oadp-short} supports backup and restore of custom security context constraints (SCC) with `ServiceAccounts` through `ClusterRole` and `ClusterRoleBinding` resources, not only those listed in the SCC user fields. This enhancement simplifies the backup and restore process of custom SCCs, as it no longer relies on the user field of the SCC. Instead, it uses the location of the SCC used by a pod and the service account associated with the pod. This results in a more reliable and secure backup and restore process for {product-title} clusters, essential for IBM Cloud Pak for Data.
```

+
link:https://redhat.atlassian.net/browse/OADP-7513[OADP-7513]

`VolumePolicy` support for bypassing pending PVCs during backups::
With this update, you can bypass unbound persistent volume claims (PVCs) during backups. This enhancement improves backup success rates, especially when PVCs are in a `Pending` state. Additionally, it expands the capabilities of volume policy actions, offering greater control and flexibility in managing backups.
+
link:https://redhat.atlassian.net/browse/OADP-6739[OADP-6739]

Optimized Data Mover performance in containerized environments with new Go version::

> With the migration to Go 1.25, the Data Mover defaults the `MaxParallelFileReads` value to the `cgroup` CPU bandwidth limit corresponding to the container's CPU limit rather than the total number of CPU cores on the node. Before this update, `MaxParallelFileReads` used the node's full CPU count unless `datamoverConfig.ParallelFilesUpload` was explicitly set in the node-agent-config map, which could lead to overuse of resources in constrained container environments. With this update, Data Mover operations automatically respect container CPU limits, improving resource efficiency and stability without requiring manual configuration changes.

Reviewer 3: ```suggestion
With the migration to Go 1.25, the Data Mover defaults the `MaxParallelFileReads` value to the `cgroup` CPU bandwidth limit corresponding to the container CPU limit rather than the total number of CPU cores on the node. Before this update, `MaxParallelFileReads` used the full CPU count of the node unless `datamoverConfig.ParallelFilesUpload` was explicitly set in the node-agent-config map, which could lead to overuse of resources in constrained container environments. With this update, Data Mover operations automatically respect container CPU limits, improving resource efficiency and stability without requiring manual configuration changes.
```

Here and at a few other places, I would avoid the possessive `'s` with inanimate objects. Just an IBM style guide thing. While it is necessary sometimes, I think we can avoid it here.

Author: I agree, thank you!

+
link:https://redhat.atlassian.net/browse/OADP-6731[OADP-6731]

`VolumeSnapshotClass` no longer required for CSI backup and restore::
The Container Storage Interface (CSI) backup and restore process no longer includes or requires the `VolumeSnapshotClass` resource. Before this update, the `VolumeSnapshotClass` was included in CSI backups and needed to be present during restore, adding an unnecessary dependency. With this update, CSI snapshot-based backups and restores complete successfully without `VolumeSnapshotClass` in the cluster, simplifying the backup resource footprint and improving restore reliability in environments where the `VolumeSnapshotClass` might not be pre-configured.
+
link:https://redhat.atlassian.net/browse/OADP-6726[OADP-6726]

Configurable cache volumes for node agent pods::

> This release introduces configurable repository cache storage locations for the node agent Data Mover and pod volume restore operations. Before this update, backup engines such as Kopia stored metadata and snapshot caches implicitly within the pod's ephemeral root file system. This often caused host node disk exhaustion, `DiskPressure` taints, and unexpected node evictions during heavy or concurrent operations.

Reviewer 3: ```suggestion
This release introduces configurable repository cache storage locations for the node agent Data Mover and pod volume restore operations. Before this update, backup engines such as Kopia stored metadata and snapshot caches implicitly within the ephemeral root file system of the pod. This often caused host node disk exhaustion, `DiskPressure` taints, and unexpected node evictions during heavy or concurrent operations.
```

+

> With this release, administrators can isolate and route repository cache files to dedicated local volumes or dynamically provisioned persistent volume claims. As a result, cache volumes protect the host node's system disk from running out of space and increases the reliability of large-scale backups and restores.

Reviewer 3: ```suggestion
With this release, administrators can isolate and route repository cache files to dedicated local volumes or dynamically provisioned persistent volume claims. As a result, cache volumes protect the system disk of the host node from running out of space and this increases the reliability of large-scale backups and restores.
```

+
link:https://redhat.atlassian.net/browse/OADP-7977[OADP-7977]

CSI snapshot early frequent polling can be configured in the DPA CR::
Before this update, a Velero change introduced 1-second Container Storage Interface (CSI) snapshot readiness polling for the first 10 seconds of every backup, designed specifically for Microsoft Volume Shadow Copy Service (VSS) workloads on Windows. {oadp-short} did not expose this behavior as a configurable option in the `DataProtectionApplication` (DPA) custom resource (CR). As a consequence, all CSI-enabled backups used the earlier polling behavior regardless of whether VSS compatibility was needed, and you had no way to disable or control this behavior through {oadp-short}. This release adds a new `enableCSISnapshotEarlyFrequentPolling` field under `configuration.velero` in the DPA CR. By default, this feature is disabled. If your workloads require this frequent polling behavior, you must enable it.
+
link:https://redhat.atlassian.net/browse/OADP-7882[OADP-7882]

Controller-manager no longer enters a `CrashLoopBackOff` loop on {sno} clusters without Console CRD::

> Before this update, the controller-manager entered a `CrashLoopBackOff` loop due to unconditional interaction with a missing `ConsoleCLIDownload` custom resource definition (CRD) on clusters without Console capability. As a consequence, you experienced pod crashes on {sno} clusters due to the missing `ConsoleCLIDownload` CRD. With this release, the {oadp-short} Operator skips CLI download setup when the Console CRD is not here. As a result, the controller-manager no longer enters a `CrashLoopBackOff` loop on {sno} clusters, improving its stability.

Reviewer 3: ```suggestion
Before this update, the controller-manager entered a `CrashLoopBackOff` loop due to unconditional interaction with a missing `ConsoleCLIDownload` custom resource definition (CRD) on clusters without Console capability. As a consequence, you experienced pod crashes on {sno} clusters due to the missing `ConsoleCLIDownload` CRD. With this release, the {oadp-short} Operator skips CLI download setup when the Console CRD is not available. As a result, the controller-manager no longer enters a `CrashLoopBackOff` loop on {sno} clusters, improving its stability.
```

+
link:https://redhat.atlassian.net/browse/OADP-7868[OADP-7868]

DPA no longer reports an error when the `podConfig` value is empty and `nodeAgent` affinity is specified::
Before this update, `DataProtectionApplication` (DPA) custom resource reported an error due to an empty `podConfig` with specified `nodeAgent` affinity. This caused DPA to fail to reconcile. With this release, DPA reports error only when `nodeAgent.podConfig.nodeSelector` does not match `spec.configuration.nodeAgent.LoadAffinityConfig` when `podConfig` is explicitly set.
+
link:https://redhat.atlassian.net/browse/OADP-7673[OADP-7673]

Backup phase `Queued` and `ReadyToStart` no longer cause reconcile issues in `nonAdminController`::
Before this update, the `Queued` and `ReadyToStart` backup phases in `nonAdminController` caused reconcile issues due to their unsupported status. With this release, the `Queued` and `ReadyToStart` no longer cause reconcile issues in `nonAdminController`.
+
link:https://redhat.atlassian.net/browse/OADP-7382[OADP-7382]

CSI `VolumeGroupSnapshot` restore no longer fails with Ceph RBD storage class::

> Before this update, restoring a `VolumeGroupSnapshot` (VGS) backed up from Ceph RBD storage failed with a `cannot find the group snapshot for VolumeSnapshot` error message. This error occurred because Velero deleted the `VolumeGroupSnapshotContent` after backup, leaving no matching resource during restore. Also the `VolumeSnapshotClassName` was deleted from restored `VolumeSnapshotContent`, preventing the Container Storage Interface (CSI) controller from authenticating to verify snapshots. With this update, Velero preserves the group snapshot handle as an annotation during backup, recreates a stub `VolumeGroupSnapshotContent` during restores, and restore the correct `VolumeSnapshotClass` reference. As a result, VGS restores with Ceph RBD complete successfully. Note that VGS is a Technology Preview feature.

Reviewer 3: ```suggestion
Before this update, restoring a `VolumeGroupSnapshot` (VGS) backed up from Ceph RBD storage failed with a `cannot find the group snapshot for VolumeSnapshot` error message. This error occurred because Velero deleted the `VolumeGroupSnapshotContent` after backup, leaving no matching resource during restore. Also the `VolumeSnapshotClassName` was deleted from restored `VolumeSnapshotContent`, preventing the Container Storage Interface (CSI) controller from authenticating to verify snapshots. With this update, Velero preserves the group snapshot handle as an annotation during backup, recreates a stub `VolumeGroupSnapshotContent` during restores, and restores the correct `VolumeSnapshotClass` reference. As a result, VGS restores with Ceph RBD complete successfully. Note that VGS is a Technology Preview feature.
```

+
link:https://redhat.atlassian.net/browse/OADP-5829[OADP-5829]

AWS S3 bucket region is auto-detected in `BackupStorageLocation` configuration::
Before this update, the AWS bucket region was missing in the `DataProtectionApplication` configuration when using S3 `BackupStorageLocation` (BSL) with `snapshotLocations`. As a consequence, backups failed due to missing region in AWS S3 bucket. With this release, AWS S3 bucket regions are automatically detected in BSL.
+
link:https://redhat.atlassian.net/browse/OADP-5777[OADP-5777]

BSL status message returns an error when an Azure storage bucket does not exist::
Before this update, when a `DataProtectionApplication` was configured with a non-existing Azure storage bucket, the `BackupStorageLocation` (BSL) validation returned a raw HTTP response in the `status.message` field. As a consequence, the BSL status was unclear. With this release, the BSL `status.message` field returns a concise, human-readable error such as `ContainerNotFound: The specified container does not exist`.
+
link:https://redhat.atlassian.net/browse/OADP-5171[OADP-5171]

{oadp-short} Data Mover restore operations honor `loadAffinity` node selection::

> Before this update, the node agent `loadAffinity` setting was  implemented only for backup operations, not for restore operations. As a consequence, when `nodeSelector` was configured in the `DataProtectionApplication` specification, the Data Mover pod could be scheduled to a node without a running node agent during a restore. This caused the restore to hang indefinitely. With this release, the `loadAffinity` configuration is applied to both backup and restore operations. As a result, Data Mover restore pods are correctly scheduled on nodes with running node agents, preventing restore hangs and `PartiallyFailed` errors.

Reviewer 3: ```suggestion
Before this update, the node agent `loadAffinity` setting was implemented only for backup operations, not for restore operations. As a consequence, when `nodeSelector` was configured in the `DataProtectionApplication` specification, the Data Mover pod could be scheduled to a node without a running node agent during a restore. This caused the restore to hang indefinitely. With this release, the `loadAffinity` configuration is applied to both backup and restore operations. As a result, Data Mover restore pods are correctly scheduled on nodes with running node agents, preventing restore hangs and `PartiallyFailed` errors.
```

Data Mover restore no longer fails when `nodeSelector` is used::
Before this update, when a `DataProtectionApplication` (DPA) custom resource was created with the `nodeSelector` field set in the `nodeAgent` parameter, `nodeSelector` was not properly handled for the Data Mover restore action. As a consequence, Data Mover restore actions partially failed because they were scheduled on a wrong node. With this release, `nodeSelector` is correctly handled for Data Mover restore actions. As a result, restore operations complete successfully when the field is configured in the DPA.
+
link:https://redhat.atlassian.net/browse/OADP-4743[OADP-4743]

Velero CLI automatically uses the CA certificate from `BackupStorageLocation`::
Before this update, running Velero containers did not store the certificate authority (CA) certificate as a file. As a consequence, you had to manually pass the `--cacert` flag with a path to a CA certificate file on every Velero command-line interface (CLI) command, or download the Velero binary and CA certificate bundle to your workstation. This made the experience inefficient in environments with custom or self-signed certificates.
+

> With this release, the Velero CLI automatically reads the CA certificate from the `BackupStorageLocation` (BSL) configuration, and gracefully falls back to the BSL's caCert if a `--cacert` flag is provided but points to an invalid path. As a result, you can run Velero CLI commands through the `oc exec` alias without manually specifying `--cacert`, significantly simplifying day-to-day backup and restore operations in environments that use custom CA certificates.

Reviewer 3: ```suggestion
With this release, the Velero CLI automatically reads the CA certificate from the `BackupStorageLocation` (BSL) configuration, and gracefully falls back to the caCert of the BSL if a `--cacert` flag is provided but points to an invalid path. As a result, you can run Velero CLI commands through the `oc exec` alias without manually specifying `--cacert`, significantly simplifying day-to-day backup and restore operations in environments that use custom CA certificates.
```

Velero no longer restores unnecessary restore-wait init containers::
Before this update, Velero did not check whether the restore-wait init container was necessary during a restore operation. As a consequence, when a namespace was backed up with CSI or native Data Mover and then was restored, Velero incorrectly injected the restore-wait init container into the application pod spec. With this release, Velero skips restoring the restore-wait init container when it is not required by the backup method used.
+
link:https://redhat.atlassian.net/browse/OADP-3759[OADP-3759]

The `defaultVolumesToFSBackup` flag matches `defaultVolumesToFsBackup`::
Before this update, the `dpa.spec.configuration.velero.defaultVolumesToFSBackup` flag used different casing than the `backup.spec.defaultVolumesToFsBackup` flag. As a consequence, the inconsistency between the two flags could cause confusion when configuring pod volume file system backup settings. With this release, the two flags are identical.
+
link:https://redhat.atlassian.net/browse/OADP-3692[OADP-3692]

{oadp-short} CLI automatically inherits the CA certificate from `BackupStorageLocation`::

> Before this update, the {oadp-short} command-line interface (CLI) did not inherit the certificate authority (CA) certificate from the `BackupStorageLocation` (BSL) when retrieving backup logs. As a consequence, you were unable to retrieve backup logs by using `oc oadp backup logs`, even though the backup itself completed successfully and the `DataProtectionApplication` (DPA) custom resource was correctly configured with a valid CA certificate. With this release, the {oadp-short} CLI automatically uses the CA certificate from the BSL when retrieving backup logs. As a result, backup log retrieval  works correctly when a valid CA certificate is present in the BSL.

Reviewer 3: ```suggestion
Before this update, the {oadp-short} command-line interface (CLI) did not inherit the certificate authority (CA) certificate from the `BackupStorageLocation` (BSL) when retrieving backup logs. As a consequence, you were unable to retrieve backup logs by using `oc oadp backup logs`, even though the backup itself completed successfully and the `DataProtectionApplication` (DPA) custom resource was correctly configured with a valid CA certificate. With this release, the {oadp-short} CLI automatically uses the CA certificate from the BSL when retrieving backup logs. As a result, backup log retrieval works correctly when a valid CA certificate is present in the BSL.
```

+
link:https://redhat.atlassian.net/browse/OADP-3039[OADP-3039]

[id="known-issues-1-6-0_{context}"]
== Known issues

`CloudStorage` bucket creation fails on {GCP} with WIF authentication::
The `CloudStorage` resource fails to detect that a {GCP} bucket does not exist and returns a `BucketCheckError` condition, preventing automatic bucket creation.
+
To work around this problem, use an existing {GCP} bucket when configuring `CloudStorage` on OpenShift clusters on {GCP} using Workload Identity Federation (WIF) authentication. As a result, the controller skips bucket creation and proceeds with the existing bucket for backup storage.
+
link:https://redhat.atlassian.net/browse/OADP-8085[OADP-8085]

> Non-admin backups of {odf-short} encrypted volumes fails with {oadp-short}::

Reviewer 3: ```suggestion
Non-admin backups of {odf-short} encrypted volumes fail with {oadp-short}::
```

+
To use this feature, enable QEMU backup on VM disks (`CBT = true`) and set up the volume policy to `virt` for target volumes.
+
link:https://redhat.atlassian.net/browse/OADP-6584[OADP-6584]

`VolumeGroupSnapshot` (Technology Preview)::
With `VolumeGroupSnapshot` (VGS) you can group and snapshot multiple persistent volume claims (PVCs) together using label-based selection. This ensures point-in-time data consistency across related volumes, improving backup reliability for stateful applications that use multiple persistent volumes. VGS is a Technology Preview feature.
+
link:https://redhat.atlassian.net/browse/OADP-6581[OADP-6581]

[id="removed-features-1-6-0_{context}"]
== Removed features

Restic backup is removed::

> Creating file system backups with Restic as an uploader type is removed and is no longer supported. Setting `uploaderType: restic` in `spec.configuration.nodeAgent.uploaderType` in the `DataProtectionApplication` (DPA) customer resource (CR) results in a reconcile error. You can create file system backups by using Kopia instead.

Reviewer 3: ```suggestion
Creating file system backups with Restic as an uploader type is removed and is no longer supported. Setting `uploaderType: restic` in `spec.configuration.nodeAgent.uploaderType` in the `DataProtectionApplication` (DPA) custom resource (CR) results in a reconcile error. You can create file system backups by using Kopia instead.
```
